src/librustc_lint/types.rs

```diff
@@ -416,7 +416,7 @@ impl<'a, 'tcx> ImproperCTypesVisitor<'a, 'tcx> {
                 }
                 match def.adt_kind() {
                     AdtKind::Struct => {
-                        if !def.repr.c() {
+                        if !def.repr.c() && !def.repr.transparent() {
                             return FfiUnsafe("found struct without foreign-function-safe \
                                               representation annotation in foreign module, \
                                               consider adding a #[repr(C)] attribute to the type");
```

Comment:
Maybe change this message to refer to `#[repr(transparent)]`?

Reply:
Possibly. The message is already too long IMHO. Maybe this lint should use a `help` diagnostic.

Reply:
I think I'll do this in a follow-up PR.